tests/test_filesystems.py

```diff
@@ -0,0 +1,344 @@
+    with pytest.raises(CreateFailed):
+        SynapseFS("syn50545516/test.txt")
+
+    # Path with no Synapse ID
+    with pytest.raises(CreateFailed):
+        SynapseFS("DCQC Test Project")
+
+    # Path that doesn't start with a Synapse ID
+    with pytest.raises(CreateFailed):
+        SynapseFS("DCQC Test Project/syn50557597")
+
+
+@pytest.mark.integration
+def test_that_a_rootless_synapsefs_can_open_a_random_file_by_id():
+    fs = SynapseFS()
+    with fs.open("syn50555279") as infile:
+        contents = infile.read()
+    assert contents == "foobar\n"
+
+
+def test_for_an_error_when_open_a_path_without_a_starting_synapse_id_while_rootless():
+    fs = SynapseFS()
+    with pytest.raises(ValueError):
+        fs._path_to_synapse_id("DCQC Test Project/syn50555279")
+
+
+def test_that_a_path_with_multiple_synapse_ids_can_be_traversed():
+    fs = SynapseFS()
+    info = fs.getinfo("syn50545516/syn50557597")
+    assert info.name == "TestSubDir"
+    assert info.is_dir
+
+
+def test_that_retrieving_the_parent_id_for_a_synapse_id_path_works():
+    fs = SynapseFS()
+    actual = fs._path_to_parent_id("syn50555279")
+    assert actual == "syn50545516"
+
+
+def test_for_an_error_when_retrieving_the_parent_id_for_a_non_synapse_id_path():
+    fs = SynapseFS()
+    with pytest.raises(ValueError):
+        fs._path_to_parent_id("test.txt")
+
+
+def test_for_an_error_when_retrieving_the_parent_entity_for_a_project():
+    fs = SynapseFS()
+    with pytest.raises(ValueError):
+        fs._get_parent_entity("syn50545516")
+
+
+def test_that_providing_an_empty_syn_url_to_open_fs_will_create_a_rootless_synapsefs():
+    fs = open_fs("syn://")
+    assert isinstance(fs, SynapseFS)
+    assert fs.root is None
+
+
+def test_for_fs_errors_when_using_synapse_errors_context_manager():
+    with pytest.raises(errors.FSError):
+        with synapse_errors("foo"):
+            raise SynapseFileNotFoundError("bar")
+    with pytest.raises(errors.FSError):
+        with synapse_errors("foo"):
+            raise SynapseHTTPError("does not exist")
+    with pytest.raises(errors.FSError):
+        with synapse_errors("foo"):
+            raise SynapseHTTPError("already exists")
+    with pytest.raises(SynapseHTTPError):
+        with synapse_errors("foo"):
+            raise SynapseHTTPError("something else")
+    with pytest.warns(UserWarning):
+        with synapse_errors("foo"):
+            raise SynapseHTTPError("File size must be at least one byte")
+
+
+def test_that_a_remote_file_without_a_close_on_callable_can_be_closed():
+    with TemporaryFile() as temp_file:
+        remote_file = RemoteFile(temp_file, "foo", "w", on_close=None)
+        remote_file.close()
+
+
+def test_that_staging_a_local_file_creates_a_copy(get_data):
+    path = get_data("test.txt")
+    local_fs = open_fs(f"osfs://{path.parent}")
+    with TemporaryDirectory() as tmp_dir_name:
+        tmp_dir_path = Path(tmp_dir_name)
+        target_path = tmp_dir_path / "test.txt"
```

Comment:
Indeed, the `Path` class should only be used for local paths. Note that the class will dynamically use `WindowsPath` or `PosixPath` depending on the OS. In your example, I'm guessing that it's interpreting the `s3` as a Windows drive, even if you're not on Windows. 

When using PyFileSystem URIs, I've been using `PurePosixPath`. I can do this because PyFileSystem expects POSIX-style paths even on Windows ([docs](https://docs.pyfilesystem.org/en/latest/concepts.html#paths)). 

The [pathlib docs](https://docs.python.org/3.11/library/pathlib.html#module-pathlib) do a good job at describing that pure paths never access the file system whereas concrete paths (like `Path`) do. In the case of URIs, pure paths make more sense since I typically just want to perform string operations on paths (like joining parts or retrieving the parent).